src/test/java/com/synopsys/integration/stepworkflow/StepWorkflowTest.java

```diff
@@ -65,238 +106,120 @@ public void testUnsuccessfulSingleStepStepWorkflow() {
-        StepWorkflow<Integer> buildStepWorkflow = successfulBuilderOfSupplierOne.build();
-        assertEquals(successfulSupplierOne, buildStepWorkflow.start.step);
-        assertEquals(successfulSupplierOne, buildStepWorkflow.end.step);
-
-        // Call run() and validate return StepWorkflowResponse
-        StepWorkflowResponse<Integer> response = successfulBuilderOfSupplierOne.run();
-        assertTrue(response.wasSuccessful());
-        assertEquals(TEST_DATA_ONE, response.getData());
-        assertNull(response.getException());
+    public void testBuilderAndSometimes() {
+        StepWorkflow.ConditionalBuilder<Integer, Integer> andSometimesConditional = successfulBuilderOfSupplierOne.andSometimes(subStepSupplierTwo);
+        StepWorkflow.Builder<Object> buildBuilder = andSometimesConditional.build(subStepFunctionOne);
+
+        assertEquals(subStepSupplierOne, buildBuilder.start.step);
+        assertEquals(subStepFunctionOne, buildBuilder.start.next.step);
+        assertNull(buildBuilder.start.next.next);
+        assertEquals(subStepFunctionOne, buildBuilder.end.step);
+        assertNull(buildBuilder.end.next);
     }
 
     @Test
-    public void testBuilderThenStep() {
-        // Create and validate then() Builder
-        StepWorkflow.Builder<Integer> thenStepWorkflow = successfulBuilderOfSupplierOne.then(successfulSupplierTwo);
-        assertEquals(successfulSupplierOne, successfulBuilderOfSupplierOne.start.step);
-        assertEquals(successfulSupplierOne, successfulBuilderOfSupplierOne.end.step);
-        assertEquals(successfulSupplierTwo, successfulBuilderOfSupplierOne.start.next.step);
-        assertNull(successfulBuilderOfSupplierOne.end.next.next);
+    public void testBuilderRun() {
+        StepWorkflowResponse<Integer> response = successfulBuilderOfSupplierOne.run();
 
-        // Call run() and validate return StepWorkflowResponse
-        StepWorkflowResponse<Integer> response = thenStepWorkflow.run();
         assertTrue(response.wasSuccessful());
-        assertEquals(TEST_DATA_TWO, response.getData());
+        assertEquals(TEST_DATA_ONE, response.getData());
         assertNull(response.getException());
     }
 
     @Test
-    public void testBuilderAndSometimesStep() {
-        // Create and validate andSometimes() Builder
-        StepWorkflow.ConditionalBuilder<Integer, Integer> andSometimesConditionalBuilder = successfulBuilderOfSupplierOne.andSometimes(successfulSupplierTwo);
-        StepWorkflow.Builder<Object> buildStepWorkflow = andSometimesConditionalBuilder.build(successfulFunctionOne);
-        assertEquals(successfulSupplierOne, buildStepWorkflow.start.step);
-        assertEquals(successfulFunctionOne, buildStepWorkflow.start.next.step);
-        assertNull(buildStepWorkflow.start.next.next);
-        assertEquals(successfulFunctionOne, buildStepWorkflow.end.step);
-        assertNull(buildStepWorkflow.end.next);
+    public void testConditionalConstructorAndThen() {
```

Comment:
The butOnlyIf test should cover what you're trying to do here, this test is likely just adding noise.

Reply:
The butOnlyIf tests don't cover then() or the Constructor for ConditionalBuilder